### PR TITLE
COP-9048: Document JSON structure

### PR DESCRIPTION
**Pull request by paulfgraham**
### Description
Storybook stories that document the structure of the JSON used for rendering forms within the COP React Form Renderer, including descriptions of what each property is for, its type, and valid values where appropriate.

There are also a number of examples, particularly for the different component types.

https://support.cop.homeoffice.gov.uk/browse/COP-9048

### CODE REVIEW

File: src/docs/Form.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import { Details, Heading, Link } from '@ukhomeoffice/cop-react-components';
+
+<!-- Local imports -->
+import { default as USER_PROFILE } from '../json/userProfile.json';
+
+<Meta title="JSON Format/Form" id="F-JSON-Form" />
+
+<Heading size="xl" id="json-form" caption="JSON Format">Form</Heading>
+
+The format of the JSON for rendering a <Link href="/?path=/docs/d-formrenderer">Form</Link>, with a sample document.
```
Comment (Ali-Baker) [inline src/docs/Form.stories.mdx:12]: The form link is not working

File: src/docs/FormTypes.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+<Meta title="JSON Format/Form types" id="F-JSON-Form-Types" />
+
+<Heading size="xl" id="json-form" caption="JSON Format">Form types</Heading>
+
+<Alert heading="Note" classModifiers="error">
+  Hub-and-spoke is the only type presently implemented. The others are works in progress.
+</Alert>
+
+The form types available in the JSON for rendering a <Link href="/?path=/docs/d-formrenderer">Form</Link>.
```
Comment (Ali-Baker) [inline src/docs/FormTypes.stories.mdx:15]: Same issue here, form link does not work

**Review by Ali-Baker — CHANGES_REQUESTED** [review]
LGTM,

Only few links not working as expected. 

Most `src/docs/FormTypes.stories.mdx` is marked as `yet to do` so I didn't comment about links not working

File: src/docs/Form.stories.mdx (hunk 1 above)
Comment (paulfgraham) [reply src/docs/Form.stories.mdx:12]: I'll add an empty story to get this working. There will be virtually nothing on it.

File: src/docs/FormTypes.stories.mdx (hunk 2 above)
Comment (paulfgraham) [reply src/docs/FormTypes.stories.mdx:15]: I'll add an empty story to get this working. There will be virtually nothing on it.

File: src/docs/Form.stories.mdx (hunk 1 above)
Comment (paulfgraham) [reply src/docs/Form.stories.mdx:12]: Done.

File: src/docs/FormTypes.stories.mdx (hunk 2 above)
Comment (paulfgraham) [reply src/docs/FormTypes.stories.mdx:15]: Done.

**Review by Ali-Baker — APPROVED** [review]

**Pull request merged**